### ISSUE TITLE
`aioboto3 Session.client()` stubs reject `botocore.config.Config`, only accept `AioConfig`

### ISSUE THREAD

**Issue by cebtenzzre** [youtype/mypy_boto3_builder#387]
## Bug

The generated `aioboto3-stubs` type `Session.client(..., config=...)` as `AioConfig | None`, but at runtime `aioboto3.Session.client()` accepts `botocore.config.Config` (the parent class). This causes a false positive from pyright:

```
No overloads for "client" match the provided arguments
```

## Reproduction

```python
import aioboto3
from botocore.config import Config

session = aioboto3.Session()
session.client('s3', config=Config(signature_version='s3v4'))
# pyright error: No overloads for "client" match the provided arguments
```

Changing `Config(...)` to `AioConfig(...)` silences the error, but plain `botocore.config.Config` works fine at runtime.

## Why it works at runtime

`aioboto3.Session` inherits `client()` from `boto3.session.Session`, which calls `self._session.create_client(...)`. The underlying `aiobotocore.session.AioSession._create_client()` accepts `config=None` with no type annotation — it just passes it through to botocore's client creation, which expects `botocore.config.Config`. Since `AioConfig` is a subclass of `botocore.client.Config`, the parameter is compatible in both directions at runtime.

## Where the stubs are too narrow

In the generated `aioboto3-stubs/session.pyi`, every `client()` overload types `config` as:

```python
config: AioConfig | None = ...
```

For example, the S3 overload (line ~6721 in `session.pyi`):

```python
@overload
def client(
    self,
    service_name: Literal["s3"],
    ...
    config: AioConfig | None = ...,    # <-- should also accept botocore.config.Config
) -> ClientCreatorContext[S3Client]: ...
```

## Expected

The `config` parameter should be typed as `botocore.config.Config | None` (which also accepts `AioConfig` since it's a subclass), or `botocore.config.Config | AioConfig | None` if you want to be explicit.

## Versions

| Package | Version |
|---------|---------|
| aioboto3 | 14.3.0 |
| aiobotocore | 2.22.0 |
| botocore | 1.37.3 |
| types-aioboto3 | 15.5.0 |
| types-aiobotocore | 3.7.0 |
| pyright | 1.1.407 |
